### ISSUE TITLE
Add Query.count to FireStore for Android, iOS

### ISSUE THREAD

**Issue by mrHerintsoaHasina** [GitLiveApp/firebase-kotlin-sdk#657]
Hi

I found missing API for aggregation queries as documented [here](https://firebase.google.com/docs/firestore/query-data/aggregation-queries#kotlin+ktx). In my case, I want to use the `count` method. But it would be nice also to include the others methods like `sum` and `average`.

**Comment by Feinate** [comment]
is there any news for this, it would be very usefull !

**Comment by chris-hatton** [comment]
Duplicate of #699

**Comment by anggrayudi** [comment]
Any update on this?

**Comment by anggrayudi** [comment]
This has shipped. `count()`, `sum()` and `average()` were all merged via #830 and are available in **v2.5.0** on Maven Central:

```kotlin
val total: Long = query.count()
val revenue: Double = query.sum("amount")
val avg: Double? = query.average("amount")
```

Implementations are present for Android, Apple and JS.

Editing my earlier note for accuracy: these are plain `suspend` functions with the aggregate source fixed to SERVER, rather than an `AggregateQuery` executed via `get(AggregateSource)`. That is a shape difference only — `AggregateSource` has exactly one constant, `SERVER` (checked against `firebase-firestore` 26.1.2), and the Apple/JS equivalents are server-only too, so no alternative source exists to expose.

This one looks closeable.

**Comment by nbransby** [comment]
There's an open PR for this: #785 by @Daeda88, which adds `AggregateQuery`, `AggregateField.Count` and `Query.count()` across Android, Apple and JS, with tests.

Note this duplicates #699 — worth consolidating once #785 lands.

**Comment by nbransby** [comment]
Closing: this is on master. `Query.count()`, `sum()` and `average()` shipped in #830 and are in 2.5.0 and later.

```kotlin
val total: Long = query.count()
val revenue: Double = query.sum("amount")
val avg: Double? = query.average("amount")
```

They are plain suspend functions against the server rather than an `AggregateQuery` you execute yourself. #785 proposes the `AggregateQuery` shape alongside a bigger SDK bump, and can carry that discussion if there is still appetite for it.

---
_Generated by [Claude Code](https://claude.ai/code)_
